Add support for using current date when logging entry w/ no date

craftyguy opened the pull request:
This adds support for automatically using today's date if the user does
not enter a date when using the 'log' function. I'm not totally happy
with the way I modified the "CheckUserEnteredProperDate" to essentially
ignore the fact the user didn't specify -d at the commandline, which was
necessary to have the program continue so that today's date could be
used later on. Definitely open to suggestions if there's a better way to
do that!

craftyguy committed:
Add support for using current date when logging entry w/ no date

This adds support for automatically using today's date if the user does
not enter a date when using the 'log' function. I'm not totally happy
with the way I modified the "CheckUserEnteredProperDate" to essentially
ignore the fact the user didn't specify -d at the commandline, which was
necessary to have the program continue so that today's date could be
used later on. Definitely open to suggestions if there's a better way to
do that!

diegomacario commented:
Thank you so much @craftyguy  ! This is such an important feature. I really felt it was missing from release 1.0.0. I will test your your implementation and hopefully merge it by tomorrow. Again, thank you!

diegomacario committed:
Include <time.h> for Windows support and limit automatic date recognition to the log command.

Automatic date recognition was affecting the tab command. It still worked properly, but when a date was not provided it would print an incorrect error message.

diegomacario committed:
Update the version of the application to 1.1.0.
